peps/pep-0733.rst

```diff
@@ -0,0 +1,634 @@
+
+Binding Generators
+------------------
+
+Libraries that create bindings between Python and other object models,
+paradigms or languages, such as
+`pybind11 <https://pybind11.readthedocs.io/en/stable/>`__ for C++11,
+`PyO3 <https://github.com/PyO3/pyo3>`__ for Rust,
+`PySide <https://pypi.org/project/PySide/>`__ for Qt,
+`PyGObject <https://pygobject.readthedocs.io/en/latest/>`__ for GTK,
+`Pygolo <https://gitlab.com/pygolo/py>`__ for Go,
+`PyJNIus <https://github.com/kivy/pyjnius/>`__ for Java, or
+`SWIG <https://swig.org/>`__ for C/C++.
+
+They need to:
+
+* Create custom objects (e.g. function/module objects
+  and traceback entries) that match the behavior of equivalent
+  Python code as closely as possible.
+* Dynamically create objects which are static in traditional
+  C extensions (e.g. classes/modules), and need CPython to manage
+  their state and lifetime.
+* Adapt foreign objects (strings, GC'd containers), with low overhead.
+* Adapt external mechanisms, execution models and guarantees to the
+  Python way (green threads/continuations, one-writer-or-multiple-readers
+  semantics, virtual multiple inheritance, 1-based indexing, super-long
+  inheritance chains, goroutines, channels, etc.).
+
+Strengths of the C API
+======================
+
+While the bulk of this document is devoted to problems with the
+C API that we would like to see fixed in any new design, it is
+also important to point out the strengths of the C API, and to
+make sure that they are preserved.
+
+As mentioned in the introduction, the C API enabled the
+development and growth of the Python ecosystem over the last
+three decades, while evolving to support use cases that it was
+not originally designed for. This track record in itself is
+indication of how effective and valuable it has been.
+
+A number of specific strengths were mentioned in the
+capi-workgroup discussions. Heap types were identified
+as much safer and easier to use than static types
+[`Issue 4 <https://github.com/capi-workgroup/problems/issues/4#issuecomment-1542324451>`__].
+
+API functions that take a C string literal for lookups based
+on a Python string are very convenient
+[`Issue 30 <https://github.com/capi-workgroup/problems/issues/30#issuecomment-1550098113>`__].
+
+The Limited API and stable ABI hide implementation details and
+make it easier to evolve Python
+[`Issue 30 <https://github.com/capi-workgroup/problems/issues/30#issuecomment-1560083258>`__].
+
+C API problems
+==============
+
+The remainder of this document summarizes and categorizes the problems that were reported on
+the `capi-workgroup <https://github.com/capi-workgroup/problems/issues/>`__ repository.
+The issues are grouped into nine categories:
+
+- API Evolution and Maintenance
+- API Specification and Abstraction
+- Object Reference Management
+- Type Definition and Object Creation
+- Error Handling
+- API Tiers and Stability Guarantees
+- Use of the C Language
+- Implementation Flaws
+- Missing Functionality
```

Comment:
```suggestion
The issues are grouped into several categories.
```
The list feels redundant with the headers (Sphinx will generate a TOC), and "nine" is prone to get out of date.